Classes/ScrnBalance.uc

```diff
@@ -3205,51 +3034,84 @@ function ServerTraveling(string URL, bool bItems)
     if (NextMutator != None)
         NextMutator.ServerTraveling(URL,bItems);
 
-    if ( ScrnGT == none || ScrnGT.ScrnGameLength == none )
+    Cleanup();
+}
+
+simulated function Destroyed()
+{
+    Cleanup();
```

Comment:
Are you sure that everything you assign in that class automatically and properly gets destroyed?

Reply:
Most map-persistent actors (GameInfo, mutators, game rules, etc.) are destroyed by Garbage Collector at map change (from native code). UnrealScript is not called at all when an actor gets destroyed natively. You can check the log - there is no "ScrnBalance destroyed" message because `ScrnBalance.Destroyed()` is never called. That's why I'm doing cleanup in `ServerTraveling()`, and there is no need to put the code elsewhere.

Here is a server log sample at map change:
```
ScrnBalance: ******************** SERVER TRAVEL ********************
ScrnBalance: New Map: KF-BioticsLab
ScrnBalance: Options: ?Game=ScrnBalanceSrv.TSCGame?Mutator=None?GameLength=40
ScriptLog: ProcessServerTravel: KF-BioticsLab?Game=ScrnBalanceSrv.TSCGame?Mutator=None?GameLength=40
ScriptLog: PreClientTravel
Log: Server switch level: KF-BioticsLab?Game=ScrnBalanceSrv.TSCGame?Mutator=None?GameLength=40
Log: Browse: KF-BioticsLab?Class=Engine.Pawn?Character=Steampunk_Demolition?team=1?Sex=M?Game=ScrnBalanceSrv.TSCGame?GameLength=40?VACSecured=true?MaxPlayers=6?Port=7707?Mutator=None?Name=[ScrN]PooSH
NetComeGo: Close TcpipConnection 127.0.0.1:54290 02/05/22 14:09:39
Exit: Socket shut down
Log: Collecting garbage
Log: Purging garbage
Log: (Karma): Level Karma Terminated.
Log: Garbage: objects: 106173->62239; refs: 810650
```

Actually, there is no need to destroy local objects (`BurnMech` etc.) at all because those will get cleaned by GC anyway. The reason why I still doing that is preventing them from accessing `class'ScrnBalance'.default.Mut`. The latter must be cleaned, otherwise it won't by garbage-collected due to having a static reference.

Reply:
Noted.